onnx/test/optimizer_test.py

```diff
@@ -99,6 +99,57 @@ def test_fuse_transpose_into_gemm(self):
         assert len(list(optimized_model.graph.node)) == 1
         assert optimized_model.graph.node[0].op_type == "Gemm"
 
+    def test_fuse_conv_add_into_bias(self):
+        conv = helper.make_node("Conv", ["X", "Y"], ["Z"])
+        add = helper.make_node("Add", ["Z", "A"], ["B"], broadcast=1, axis=1)
+        graph = helper.make_graph(
+            [conv, add],
+            "test",
+            [helper.make_tensor_value_info("X", TensorProto.FLOAT, (1, 5, 3, 3)),
+             helper.make_tensor_value_info("Y", TensorProto.FLOAT, (5, 16, 3, 3)),
+             helper.make_tensor_value_info("A", TensorProto.FLOAT, (16,))],
+            [helper.make_tensor_value_info("B", TensorProto.FLOAT, (1, 16, 1, 1))],
+            value_info=[
+                helper.make_tensor_value_info("Z", TensorProto.FLOAT, (1, 16, 1, 1)),
+            ]
+        )
+        optimized_model = self._optimized(graph, ["fuse_conv_add_into_bias"])
+
+        assert len(list(optimized_model.graph.node)) == 1
+
+    def test_fuse_conv_add_into_bias_no_fuse_3d_bias(self):
+        conv = helper.make_node("Conv", ["X", "Y"], ["Z"])
+        add = helper.make_node("Add", ["Z", "A"], ["B"], broadcast=1, axis=1)
+        graph = helper.make_graph(
+            [conv, add],
+            "test",
+            [helper.make_tensor_value_info("X", TensorProto.FLOAT, (1, 5, 3, 3)),
+             helper.make_tensor_value_info("Y", TensorProto.FLOAT, (5, 16, 3, 3)),
+             helper.make_tensor_value_info("A", TensorProto.FLOAT, (16, 1, 1))],
+            [helper.make_tensor_value_info("B", TensorProto.FLOAT, (1, 16, 1, 1))],
+            value_info=[
+                helper.make_tensor_value_info("Z", TensorProto.FLOAT, (1, 16, 1, 1)),
+            ]
+        )
+        optimized_model = self._optimized(graph, ["fuse_conv_add_into_bias"])
+
+        assert len(list(optimized_model.graph.node)) == 2
+
+    def test_fuse_conv_add_into_bias_no_fuse_no_size(self):
+        conv = helper.make_node("Conv", ["X", "Y"], ["Z"])
+        add = helper.make_node("Add", ["Z", "A"], ["B"], broadcast=1, axis=1)
+        graph = helper.make_graph(
+            [conv, add],
+            "test",
+            [helper.make_tensor_value_info("X", TensorProto.FLOAT, (1, 5, 3, 3)),
+             helper.make_tensor_value_info("Y", TensorProto.FLOAT, (5, 16, 3, 3)),
+             helper.make_tensor_value_info("A", TensorProto.FLOAT, (16,))],
```

Comment:
Aha, I know why do you want to check weight dim.
I think checking weight dim should not be included here. It's a Conv specification unconformity. Should check in checker.py? maybe.

Reply:
I think we can depends on weight's shape information. Because in the operator doc, we explicitly say both weight_shape[0].dim = bias_shape[0].dim = M.